lib/rbvmomi/connection.rb

```diff
@@ -136,7 +136,7 @@ def obj2xml xml, name, type, is_array, o, attrs={}
             obj2xml xml, name, expected.wsdl_name, false, e, attrs
           end
         end
-      when BasicTypes::ManagedObject
+      when BasicTypes::ManagedObject, BasicTypes::OpaqueObject
```

Comment:
This this line could check for the mixin?

Reply:
They have to have different parent classes so not sure that we would move this to checking a single item.  If anything I think moving the serialization/deserialization into the object instead of here (there is a `# hic sunt dracones` comment on this method after all)